# Checking frauds 01-19 to 03-19

In [1]:
import os

import pandas as pd

In [2]:
# configs
pd.set_option("display.max_columns", 500)
pd.set_option("display.max_rows", 500)

## Loading Datasets

In [6]:
# validation set (01-03 2019)

filename = "../../../data/split/1.1 - df_val 01-19to03-19.feather"

# check if file exists
if not os.path.isfile(filename):
    raise Exception(f"File {filename} doesn't exists")

In [5]:
df = pd.read_feather(filename)
df.head()

/home/voolkia/anaconda3/lib/python3.6/site-packages/pyarrow/pandas_compat.py:752: FutureWarning: .labels was deprecated in version 0.24.0. Use .codes instead.
  labels, = index.labels


NUM_SECU_EXPED  cond_04  cond_05  cond_06  cond_09  cond_11  cond_12  \
0   1905896179999     -1.0     -1.0     -1.0     -1.0     -1.0     -1.0   
1   1905896259999     -1.0     -1.0     -1.0     -1.0     -1.0     -1.0   
2   1905896279999     -1.0     -1.0     -1.0     -1.0     -1.0     -1.0   
3   1905896299999     -1.0     -1.0     -1.0     -1.0     -1.0     -1.0   
4   1905896309999     -1.0      0.0      0.0      0.0      0.0     -1.0   

   cond_32  cond_37  cond_C1  cond_C10  cond_C11  cond_C12  cond_C13  \
0     -1.0     -1.0     -1.0      -1.0      -1.0      -1.0       0.0   
1     -1.0     -1.0      0.0      -1.0      -1.0      -1.0      -1.0   
2      0.0     -1.0     -1.0       0.0       0.0       0.0       0.0   
3     -1.0     -1.0     -1.0      -1.0      -1.0      -1.0       0.0   
4     -1.0     -1.0      0.0      -1.0      -1.0      -1.0       0.0   

   cond_C14  cond_C15  cond_C16  cond_C17  cond_C18  cond_C19  cond_C2  \
0      -1.0       0.0       0.0      -1.0      -1.0      -1.0     -1.0   
1      -1.0       0.0       0.0      -1.0      -1.0      -1.0     -1.0   
2      -1.0       0.0       0.0      -1.0       0.0       0.0     -1.0   
3      -1.0       0.0       0.0      -1.0      -1.0      -1.0     -1.0   
4      -1.0       0.0       0.0       0.0      -1.0      -1.0     -1.0   

   cond_C20  cond_C21  cond_C3  cond_C4  cond_C5  cond_C6  cond_C7  cond_C8  \
0      -1.0      -1.0     -1.0     -1.0     -1.0     -1.0     -1.0     -1.0   
1      -1.0      -1.0     -1.0     -1.0     -1.0      0.0      0.0      0.0   
2       0.0      -1.0     -1.0     -1.0     -1.0     -1.0     -1.0     -1.0   
3      -1.0      -1.0     -1.0     -1.0     -1.0     -1.0     -1.0     -1.0   
4      -1.0       0.0     -1.0     -1.0     -1.0      0.0     -1.0      0.0   

   cond_C9  total_condicion  es_gte_5  EXISTE_FRAUDE  EXISTE_INHABILITAR  \
0     -1.0              0.0     False          False                  -1   
1      0.0              0.0     False          False                  -1   
2     -1.0              0.0     False           True                   0   
3     -1.0              0.0     False          False                  -1   
4      0.0              0.0     False          False                  -1   

   preg_1  preg_2  preg_3  preg_4  preg_5  preg_6  preg_7  preg_8  preg_9  \
0      -1      -1      -1      -1      -1      -1      -1      -1      -1   
1      -1      -1      -1      -1      -1      -1      -1      -1      -1   
2      -1      -1      -1      -1      -1      -1      -1      -1      -1   
3      -1      -1      -1      -1      -1      -1      -1      -1      -1   
4      -1      -1      -1      -1       1       0      -1      -1       0   

   preg_10  preg_11  preg_12  preg_15  preg_16  preg_18  preg_19  preg_22  \
0       -1       -1       -1       -1       -1       -1       -1       -1   
1       -1       -1       -1       -1       -1       -1       -1       -1   
2       -1       -1       -1       -1       -1       -1       -1       -1   
3       -1       -1       -1       -1       -1       -1       -1       -1   
4       -1        0       -1       -1       -1       -1       -1       -1   

   preg_24  preg_27  preg_28  preg_30  preg_31  preg_32  preg_33  preg_34  \
0       -1       -1       -1       -1       -1       -1       -1       -1   
1       -1       -1       -1       -1       -1       -1       -1       -1   
2       -1       -1       -1       -1       -1       -1       -1       -1   
3       -1       -1       -1       -1       -1       -1       -1       -1   
4       -1       -1       -1       -1       -1       -1       -1       -1   

   preg_36  preg_37  CATASTROFICO  CERRADURA_BAUL  CERRADURA_DERECHA  \
0       -1       -1             0               0                  0   
1       -1       -1             0              -1                 -1   
2       -1       -1             0               0                  0   
3       -1       -1             0               0                

In [11]:
# validation info
df_val = pd.read_csv("fraudes eneAmar19.csv", sep=";")
df_val.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 534 entries, 0 to 533
Data columns (total 2 columns):
NUM_SECU_EXPED      534 non-null int64
FECHA_INI_INVEST    534 non-null object
dtypes: int64(1), object(1)
memory usage: 8.4+ KB


In [13]:
df_val["FECHA_INI_INVEST"] = pd.to_datetime(df_val["FECHA_INI_INVEST"], format="%d/%m/%Y")
df_val.head()

NUM_SECU_EXPED FECHA_INI_INVEST
0   1905900369999       2019-01-09
1   1908540579999       2019-01-15
2   1906026909999       2019-01-10
3   1906924089999       2019-01-07
4   1910661109999       2019-01-16

## comparing number of frauds

In [7]:
df["EXISTE_FRAUDE"].value_counts(dropna=False)

False    106741
True        523
Name: EXISTE_FRAUDE, dtype: int64

In [15]:
len(df_val)

534

In [17]:
len(df[df["EXISTE_FRAUDE"] == True]) - len(df_val)

-11

**11 frauds of difference**

## getting the missing frauds

In [18]:
our_frauds = set(df[df["EXISTE_FRAUDE"] == True]["NUM_SECU_EXPED"])
valid = set(df_val["NUM_SECU_EXPED"])

diff = valid - our_frauds
print(len(diff))
print(diff)

12
{1915283399999, 1921556889999, 1906050549999, 1935937769999, 1917364009999, 1941381749999, 1915195589999, 1926519149999, 1910232169999, 1939199849999, 1938578099999, 1910619219999}


## Analyzing missing cases

In [34]:
df_val[df_val["NUM_SECU_EXPED"].isin(diff)]

NUM_SECU_EXPED FECHA_INI_INVEST
9     1910232169999       2019-01-16
21    1915195589999       2019-02-04
100   1921556889999       2019-05-08
130   1939199849999       2019-04-09
190   1938578099999       2019-03-29
225   1917364009999       2019-02-22
255   1941381749999       2019-04-11
294   1926519149999       2019-02-28
336   1910619219999       2019-01-16
353   1915283399999       2019-05-21
389   1935937769999       2019-03-21
404   1906050549999       2019-01-10

In [38]:
df[df["NUM_SECU_EXPED"].isin(diff)][["NUM_SECU_EXPED", "EXISTE_FRAUDE"]]

NUM_SECU_EXPED  EXISTE_FRAUDE
936      1906050549999          False
8737     1910232169999          False
9640     1910619219999          False
22583    1915195589999          False
23486    1915283399999          False
35195    1917364009999          False
45048    1921556889999          False
57459    1926519149999          False
82164    1935937769999          False
97054    1938578099999          False
100469   1939199849999          False
103907   1941381749999          False

In [39]:
# there is 1 exped that we have as fraud but it's not.
our_frauds - valid

{1922289749999}

## comparing against the original dataset (investigaciones)

In [40]:
# validation set (01-03 2019)

filename = "../../../data/raw/INVESTIGACION.csv"

# check if file exists
if not os.path.isfile(filename):
    raise Exception(f"File {filename} doesn't exists")

In [42]:
df_inv = pd.read_csv(filename)
df_inv.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22468 entries, 0 to 22467
Data columns (total 43 columns):
COD_CIA                22468 non-null int64
COD_RAMO               22468 non-null int64
NUM_SINI               22468 non-null int64
NRO_EXPED              22468 non-null int64
NRO_ORDEN_INV          22468 non-null int64
NRO_INVEST             22468 non-null int64
COD_TALLER             0 non-null float64
IMPORTE                22467 non-null object
FECHA_INI_INVEST       22468 non-null object
FECHA_ENT_INVEST       22468 non-null object
FECHA_FIN_INVEST       13711 non-null object
FECHA_MOD_INVEST       14103 non-null object
EXISTE_INVEST          22468 non-null object
COD_INVEST_INT         129 non-null object
COD_DOCUM_INVEST1      17431 non-null object
COD_INVEST1            17431 non-null float64
NOM_INVEST1            17431 non-null object
COD_DOCUM_INVEST2      4 non-null object
COD_INVEST2            4 non-null float64
NOM_INVEST2            4 non-null object
COD_DOCUM_IN

/home/voolkia/anaconda3/lib/python3.6/site-packages/IPython/core/interactiveshell.py:2785: DtypeWarning: Columns (24) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


### the missing NUM_SECU_EXPED

In [43]:
# checking the missing value
df_inv[df_inv["NUM_SECU_EXPED"] == 1922289749999]

COD_CIA  COD_RAMO     NUM_SINI  NRO_EXPED  NRO_ORDEN_INV  NRO_INVEST  \
19581        1         1  53309325110          1              0       36028   
20200        1         1  53309325110          1              2       36028   
20269        1         1  53309325110          1              1       36028   

       COD_TALLER    IMPORTE FECHA_INI_INVEST FECHA_ENT_INVEST  \
19581         NaN  134600,00        28/2/2019        15/3/2019   
20200         NaN  134600,00        28/2/2019        15/3/2019   
20269         NaN  134600,00        28/2/2019        15/3/2019   

      FECHA_FIN_INVEST FECHA_MOD_INVEST EXISTE_INVEST COD_INVEST_INT  \
19581              NaN              NaN             S            NaN   
20200         8/4/2019         8/4/2019             S            NaN   
20269        22/3/2019        22/3/2019             S            NaN   

      COD_DOCUM_INVEST1   COD_INVEST1                    NOM_INVEST1  \
19581                CT  3.071152e+10  GSA GLOBAL SECURITY ADVISERS,   
20200                CT  3.071152e+10  GSA GLOBAL SECURITY ADVISERS,   
20269                CT  3.071152e+10  GSA GLOBAL SECURITY ADVISERS,   

      COD_DOCUM_INVEST2  COD_INVEST2 NOM_INVEST2  COD_DOCUM_INVEST3  \
19581               NaN          NaN         NaN                NaN   
20200               NaN          NaN         NaN                NaN   
20269               NaN          NaN         NaN                NaN   

       COD_INVEST3  NOM_INVEST3  COD_FRAUDE OBSERVACIONES  \
19581          NaN          NaN         NaN           NaN   
20200          NaN          NaN         0.0           NaN   
20269          NaN          NaN        20.0           NaN   

                                             COMENTARIOS  \
19581  ) VH CON 220000 KM, DICE QUE  EN UN VIAJE A SA...   
20200  ) VH CON 220000 KM, DICE QUE  EN UN VIAJE A SA...   
20269  ) VH CON 220000 KM, DICE QUE  EN UN VIAJE A SA...   

                                            CONCLUSIONES  NUM_SECU_SINI  \
19581                                                NaN  1922289139999   
20200  RECHAZAR POR INCUMPLIMIENTO DE LA CARGA DEL AS...  1922289139999   
20269  RECHAZAR POR INCUMPLIMIENTO DE LA CARGA DEL AS...  1922289139999   

       NUM_SECU_EXPED FECHA_EQUIPO   COD_USR  COD_ABOGADO FECHA_VTO_PLAZO  \
19581   1922289749999    28/2/2019  RODRIGAP          NaN       14/3/2019   
20200   1922289749999     8/4/2019  RODRIGAP          NaN       14/3/2019   
20269   1922289749999    22/3/2019  RODRIGAP          NaN       14/3/2019   

      CAUSA_DERIVACION MOTIVO_DERIVACION EXISTE_FRAUDE EXISTE_INHABILITAR  \
19581             PART               DAS             N                  N   
20200             PART               DAS             N                  N   
20269             PART               DAS             S                  N   

       COD_AGEN_DERIV CESVI_INF_OK IMPORTE_AHORRADO SUB_COD_FRAUDE  \
19581             NaN            N             0,00            NaN   
20200             NaN            N             0,00            NaN   
20269             NaN            N             0,00            NaN   

       TIPO_ABOGADO FECHA_DESISTIMIENTO  
19581           NaN                 NaN  
20200           NaN                 NaN  
20269           NaN                 NaN

**the NUM_SECU_EXPED number 1922289749999, it's identify as fraud in the record id 20269.**

### our 11 false fraud

In [47]:
df_inv[df_inv["NUM_SECU_EXPED"].isin(diff)][["NUM_SECU_EXPED", "EXISTE_FRAUDE"]].sort_values(by="NUM_SECU_EXPED")

NUM_SECU_EXPED EXISTE_FRAUDE
17300   1906050549999             S
17685   1906050549999             N
17849   1910232169999             S
17916   1910232169999             N
17876   1910619219999             S
17877   1910619219999             S
17926   1910619219999             N
18443   1915195589999             N
19076   1915195589999             N
21977   1915283399999             N
19455   1917364009999             N
19118   1917364009999             S
21810   1921556889999             N
19387   1926519149999             N
19258   1926519149999             S
20227   1935937769999             N
20628   1938578099999             S
20645   1938578099999             N
20770   1939199849999             S
20893   1939199849999             N
20640   1941381749999             S
20864   1941381749999             N

**It were fraud, but they weren't identified in the last record.**